### PR TITLE
EZP-27609: Update doc on Doctrine for Field Types external storage

### PR DESCRIPTION
**Pull request by DominikaK**
https://jira.ez.no/browse/EZP-27609

### CODE REVIEW

**Comment by DominikaK** [comment]
ping @alongosz

File: docs/api/field_type_api_and_best_practices.md [hunk 1]
```diff
@@ -613,21 +613,22 @@ Here is an example for **ezurl** field type:
...
     ezpublish.fieldType.ezurl.externalStorage:
         class: %ezpublish.fieldType.ezurl.externalStorage.class%
+        arguments:
+            - "@ezpublish.fieldType.ezurl.storage_gateway"
         tags:
             - {name: ezpublish.fieldType.externalStorageHandler, alias: ezurl}
 ```
 
-The configuration is straight forward. Nothing specific except the **`ezpublish.fieldType.externalStorageHandler `** tag, the `alias` attribute still begin the *fieldTypeIdentifier*.
+The configuration requires providing the **`ezpublish.fieldType.externalStorageHandler `** tag, with the `alias` attribute being the *fieldTypeIdentifier*. You also have to inject the gateway in the `arguments` tag, [see below](#gateway-based-storage_1).
```
Comment (alongosz) [inline docs/api/field_type_api_and_best_practices.md]: I wouldn't call `arguments` key a "tag". Tag is a keyword in Symfony DI Container domain. Either "pass a gateway service definition as an argument" or just remove word "tag" (to me, sounds good w/o it as well).

Comment (alongosz) [inline docs/api/field_type_api_and_best_practices.md]: `#gateway-based-storage-1`, right?

File: docs/api/field_type_api_and_best_practices.md [hunk 2]
```diff
@@ -613,21 +613,22 @@ Here is an example for **ezurl** field type:
...
+External storage configuration for basic field types is located in [eZ/Publish/Core/settings/fieldtype_external_storages.yml](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/settings/fieldtype_external_storages.yml).
 
 ##### Gateway based storage
 
-As stated in the [Field Type best practices](#gateway-based-storage), in order to be storage agnostic and external storage handler should use a *storage gateway*. This can be done by implementing another service implementing `eZ\Publish\Core\FieldType\StorageGateway` and being tagged as `ezpublish.fieldType.externalStorageHandler.gateway`.
+As stated in the [Field Type best practices](#gateway-based-storage), in order to be storage agnostic and external storage handler should use a *storage gateway*. This can be done using another service implementing `eZ\Publish\SPI\FieldType\StorageGateway`, to be injected into external storage handler ([see above](#external-storage)).
 
 **Storage gateway for ezurl**
```
Comment (alongosz) [inline docs/api/field_type_api_and_best_practices.md]: The yml code below this, which I'm unable to review here, also needs to be changed. Please see [the proper yml](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/settings/storage_engines/legacy/external_storage_gateways.yml#L41-L43). Doc further refers to connection argument. Also tagging disappeared as is no longer needed (gateway gets injected in the code mentioned above).

**Pull request merged**